src/libsyntax/parse/lexer/mod.rs

```diff
@@ -799,7 +804,18 @@ impl<'a> StringReader<'a> {
                             'n' | 'r' | 't' | '\\' | '\'' | '"' | '0' => true,
                             'x' => self.scan_byte_escape(delim, !ascii_only),
                             'u' if self.curr_is('{') => {
-                                self.scan_unicode_escape(delim)
+                            let valid = self.scan_unicode_escape(delim);
+                            if valid && ascii_only {
+                                self.err_span_(
+                                    escaped_pos,
+                                    self.last_pos,
+                                    "Unicode escape sequences cannot be used as byte or in \
+                                    byte string."
```

Comment:
Error messages in Rust conventionally do not start with an uppercase letter and also don't end with a period